src/sql/src/query_model/hir.rs

```diff
@@ -129,6 +129,88 @@ impl FromHir {
+                let group_box_id = self
+                    .model
+                    .make_box(BoxType::Grouping(Grouping { key: Vec::new() }));
+                let select_q_id =
+                    self.model
+                        .make_quantifier(QuantifierType::Foreach, select_id, group_box_id);
+                let mut key = Vec::new();
+                for k in group_key.into_iter() {
+                    // Make sure the input select box projects the source column needed
+                    let col_ref = BoxScalarExpr::ColumnReference(ColumnReference {
+                        quantifier_id: input_q_id,
+                        position: k,
+                    });
+                    let position = self
+                        .model
+                        .get_mut_box(select_id)
+                        .add_column_if_not_exists(col_ref);
+                    // Reference of the column projected by the input select box
+                    let col_ref = BoxScalarExpr::ColumnReference(ColumnReference {
+                        quantifier_id: select_q_id,
+                        position,
+                    });
+                    // Add it to the grouping key and to the projection of the
+                    // Grouping box
+                    key.push(Box::new(col_ref.clone()));
+                    self.model.get_mut_box(group_box_id).columns.push(Column {
+                        expr: col_ref,
+                        alias: None,
+                    });
+                }
+                for aggregate in aggregates.into_iter() {
+                    // Any computed expression in the pass as an argument of an aggregate
+                    // function is computed by the input select box.
+                    let input_expr = self.generate_expr(*aggregate.expr, select_id);
+                    let position = self
+                        .model
+                        .get_mut_box(select_id)
+                        .add_column_if_not_exists(input_expr);
+                    // Reference of the column projected by the input select box
+                    let col_ref = BoxScalarExpr::ColumnReference(ColumnReference {
+                        quantifier_id: select_q_id,
+                        position,
+                    });
+                    // Add the aggregate expression to the projection of the Grouping
+                    // box
+                    let aggregate = BoxScalarExpr::Aggregate {
+                        func: aggregate.func.into_expr(),
+                        expr: Box::new(col_ref),
+                        distinct: aggregate.distinct,
+                    };
+                    self.model.get_mut_box(group_box_id).columns.push(Column {
+                        expr: aggregate,
+                        alias: None,
+                    });
+                }
+
+                // Update the key of the grouping box
```

Comment:
Should we also add the group by columns to the box's `unique_keys` field?

Reply:
I was thinking that could be done as part of #9180. We would add a new `update_computed_properties` or whatever, that updates those computed properties like the `unique_keys` that would have to be invoked after any mutation to the box, including its creation. In that method, we would know how to set the `unique_keys` properly for every type of box.